src/PostgREST/DbRequestBuilder.hs

```diff
@@ -35,7 +35,7 @@ import           Protolude                hiding (from, dropWhile, drop)
 import           Text.Regex.TDFA         ((=~))
 import           Unsafe                  (unsafeHead)
 
-readRequest :: Maybe Integer -> [Relation] -> [(Text, Text)] -> ApiRequest -> Either Response ReadRequest
+readRequest :: Maybe Integer -> [Relation] -> [(Text, RetType)] -> ApiRequest -> Either Response ReadRequest
 readRequest maxRows allRels allProcs apiRequest  =
```

Comment:
I refactored the ```allProcs``` to ```HashMap Text ProcDescription``` but it didn't help to reduce the tName logic, I thought of working that by adding a ``` [Table] ``` param to ```accesibleProcs``` but the refactor will be larger. I'm going to work on the binary output for procs next, maybe It'll be better to continue the refactor on that PR.